egs/wsj/s5/utils/mkgraph.sh

```diff
@@ -79,35 +79,41 @@ P=$(tree-info $tree | grep "central-position" | cut -d' ' -f2) || { echo "Error
   echo "$0: WARNING: chain models need '--self-loop-scale 1.0'";
 
 mkdir -p $lang/tmp
+trap "rm -f $lang/tmp/LG.fst.$$" EXIT HUP INT PIPE TERM
 # Note: [[ ]] is like [ ] but enables certain extra constructs, e.g. || in
 # place of -o
 if [[ ! -s $lang/tmp/LG.fst || $lang/tmp/LG.fst -ot $lang/G.fst || \
       $lang/tmp/LG.fst -ot $lang/L_disambig.fst ]]; then
   fsttablecompose $lang/L_disambig.fst $lang/G.fst | fstdeterminizestar --use-log=true | \
     fstminimizeencoded | fstpushspecial | \
-    fstarcsort --sort_type=ilabel > $lang/tmp/LG.fst || exit 1;
-  fstisstochastic $lang/tmp/LG.fst || echo "[info]: LG not stochastic."
+    fstarcsort --sort_type=ilabel > $lang/tmp/LG.fst.$$ || exit 1;
+  fstisstochastic $lang/tmp/LG.fst.$$ || echo "[info]: LG not stochastic."
+  mv $lang/tmp/LG.fst.$$ $lang/tmp/LG.fst
```

Comment:
I'd do the move before the 'fstisstochastic' step.. this will make the printed messages easier to interpret.